push/src/main/java/org/aerogear/mobile/push/PushService.java

```diff
@@ -187,17 +234,7 @@ public void onException(Exception error) {
      * @param callback A callback to handle successful or failed unregistration
      */
     public void unregisterDevice(final Callback callback) {
```

Comment:
Can you refactor to return a Request instead of taking a callback?  I can help you with this.

Reply:
I have created a new Jira for it => https://issues.jboss.org/browse/AGDROID-806